lookout/style/typos_checker/README.md

```diff
@@ -1,53 +1,86 @@
 
-PR and issues are welcomed ;)
+# How to run it
 
-## how to reproduce
+You can follow these steps to reproduce the demo, as it was seen during the presentation of the hackathon project.
+
+PR and issues are very welcomed ;)
+
+## Run lookout
+
+You will need to have [Go](https://golang.org/doc/install#install) and [docker-compose](https://docs.docker.com/compose/install/)
+
+### 1. Install [lookout](https://github.com/src-d/lookout)
 
-### 1. Install [lookout](https://github.com/src-d/lookout) locally
 ```shell
-$ go get github.com/src-d/lookout
+$ go get github.com/src-d/lookout/...
 ```
 
-### 2. From lookout directory, let's run the app dependencies using `docker compose`
+### 2. Run lookout dependencies using `docker compose`
+
+**lookout** requires a running [bblfsh](https://doc.bblf.sh) server and a [Postgresql](https://www.postgresql.org/) database; to get them you can run from **lookout** directory:
+
 ```shell
 $ docker-compose up bblfsh postgres
 ```
-and initialize lookout database
+
+and then initialize **lookout** database
+
 ```shell
 $ lookoutd migrate
 ```
 
 ### 3. Run lookout server
+
 ```shell
 $ lookoutd serve --github-token <gh-token> --github-user <user> <repository>
 ```
+
 where:
 - `user` is the github user handler that will post the comment,
 - `gh-token` is [the token of the user](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line) that will post the comment,
 - `repository` is the repository url that will be watched for new PRs (it MUST be public)
 
-### 4. Change analyzer `config.yml`; exposed port should be `10302`
+## Run Style Analyzer
+
+You will need to have [Python](https://www.python.org/downloads/) (between `3.5` and `3.7`)
+
+### 1. Install style-analyzer
+
+Download the **Style Analyzer** repsitory, containing the `typos-checker` module, that can be found at [`irinakhismatullina/style-analyzer` at `typos-analyzer` branch](https://github.com/irinakhismatullina/style-analyzer/tree/feature/typos-analyzer)
+
+And install its dependencies, running from its folder:
+
+```shell
+$ sudo pip3 install -e .
+```
+
+### 2. Configure and run the analyzer
+
+Change the analyzer [`config.yml`](https://github.com/irinakhismatullina/style-analyzer/blob/feature/typos-analyzer/config.yml) to expose the `10302` port.
+
+And run it from **Style Analyzer** directory:
 
-### 5. Install style-analyzer as it can be found at [irinakhismatullina/style-analyzer::typos-analyzer](https://github.com/irinakhismatullina/style-analyzer/tree/feature/typos-analyzer)
 ```shell
 $ python3 -m lookout run lookout.style.typos_checker -c config.yml
 ```
 
-### 6. Open a PR with a Java file, containing some typos in some code comments
+## Test it with an example in GitHub
+
+Open a PR &ndash;into the watched repository&ndash; with a Java file, containing some typos in some code comments, example:
```

Comment:
Open a PR &ndash;into the watched repository&ndash; with a Java file, containing some typos in some code comments, for example: